controller/tasks_manager.go

```diff
@@ -47,16 +46,9 @@ type TasksManager struct {
 }
 
 // NewTasksManager configures a new tasks manager
-func NewTasksManager(conf *config.Config, state state.Store) (*TasksManager, error) {
-
+func NewTasksManager(conf *config.Config, state state.Store, watcher templates.Watcher) (*TasksManager, error) {
```

Comment:
Now that the inputs are expanding, do you think this warrants a constructor config object? I think 3 is okay, but if we add more it might be worth it